### PR TITLE
Fix more RealFields and Attempt f64+f32 Parenting [Here be dragons]

### PR DESCRIPTION
**Pull request by topokel**
My f64+f32 parenting system changes feel Dumb

Test Failures:
- `transform::components::transform::tests::test_mul` - precision error
- `transform::systems::tests::basic` - failure when indexing Transform in storage(???)
- `transform::systems::tests::parent_after` - failure when indexing Transform in storage(???)
- `transform::systems::tests::parent_before` - failure when indexing Transform in storage(???)
- `transform::systems::tests::parent_removed` - failure when indexing Transform in storage(???)
- `transform::systems::tests::zeroed` - failure when indexing Transform in storage(???)

### CODE REVIEW

**Comment by topokel** [comment]
Ok so now all the test failures are numerical

**Comment by topokel** [comment]
The only failing test left! 

```
---- transform::systems::tests::parent_after stdout ----
thread 'transform::systems::tests::parent_after' panicked at 'assertion failed: `(left == right)`
  left: `Matrix { data: [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0] }`,
 right: `Matrix { data: [0.6666666, 0.3333333, -0.6666666, 0.0, 0.3333333, 0.6666666, 0.6666666, 0.0, 0.6666666, -0.6666666, 0.3333333, 0.0, 5.0, 5.0, 5.0, 1.0] }`', amethyst_core\src\transform\systems.rs:412:13
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys_common::backtrace::_print
             at src\libstd\sys\windows\backtrace/mod.rs:94
             at src\libstd\sys\windows\backtrace/mod.rs:81
             at src\libstd\sys_common/backtrace.rs:70
   1: std::panicking::default_hook::{{closure}}
             at src\libstd\sys_common/backtrace.rs:58
             at src\libstd/panicking.rs:200
   2: std::panicking::default_hook
             at src\libstd/panicking.rs:209
   3: std::panicking::rust_panic_with_hook
             at src\libstd/panicking.rs:478
   4: std::panicking::continue_panic_fmt
             at src\libstd/panicking.rs:385
   5: std::panicking::begin_panic_fmt
             at src\libstd/panicking.rs:340
   6: amethyst_core::transform::systems::tests::parent_after
             at amethyst_core\src\transform/systems.rs:412
   7: amethyst_core::transform::systems::tests::parent_after::{{closure}}
             at amethyst_core\src\transform/systems.rs:377
   8: core::ops::function::FnOnce::call_once
             at /rustc/91856ed52c58aa5ba66a015354d1cc69e9779bdf\src\libcore\ops/function.rs:231
   9: <F as alloc::boxed::FnBox<A>>::call_box
             at src\libtest/lib.rs:1497
             at /rustc/91856ed52c58aa5ba66a015354d1cc69e9779bdf\src\libcore\ops/function.rs:231
             at /rustc/91856ed52c58aa5ba66a015354d1cc69e9779bdf\src\liballoc/boxed.rs:749
  10: _rust_maybe_catch_panic
             at src\libpanic_unwind/lib.rs:87
  11: test::run_test::run_test_inner::{{closure}}
             at /rustc/91856ed52c58aa5ba66a015354d1cc69e9779bdf\src\libstd/panicking.rs:276
             at /rustc/91856ed52c58aa5ba66a015354d1cc69e9779bdf\src\libstd/panic.rs:388
             at src\libtest/lib.rs:1452
```

File: amethyst_core/src/transform/systems.rs [hunk 1]
```diff
@@ -78,17 +99,37 @@ impl<'a> System<'a> for TransformSystem {
...
+            &self.local_modified,
+            &mut double_locals,
+            !&parents,
+        )
+            .join()
+        {
+            modified.push(entity.id());
+            local.global_matrix = local.matrix();
+
             debug_assert!(
```
Comment (AnneKitsune) [inline amethyst_core/src/transform/systems.rs:119]: Duplicated assert.

File: amethyst_core/src/transform/systems.rs [hunk 2]
```diff
@@ -37,9 +40,13 @@ impl<'a> System<'a> for TransformSystem {
         Entities<'a>,
         ReadExpect<'a, ParentHierarchy>,
         WriteStorage<'a, Transform<f32>>,
+        WriteStorage<'a, Transform<f64>>,
```
Comment (AnneKitsune) [inline amethyst_core/src/transform/systems.rs:43]: No you can't have both

**Review by AnneKitsune — CHANGES_REQUESTED** [review]
TransformSystem should be generic over N.
Also please rollback rustfmt changes.

Thanks

**Comment by topokel** [comment]
Problem with TransformSystem parameterized over the Transform number type is that it's then impossible to mix types within the hierarchy, is that ok?

**Comment by AnneKitsune** [comment]
Yes they shouldn't be mixed. That would cause all sorts of weird issues.

**Pull request closed without merging**